x-pack/plugins/alerting/server/saved_objects/transform_rule_for_export.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObject } from 'kibana/server';
+import { AuditLogger } from '../../../security/server';
+import { AlertAuditAction, alertAuditEvent } from '../alerts_client/audit_events';
+import { RawAlert } from '../types';
+
+export function transformRulesForExport(
+  rules: SavedObject[],
+  auditLogger?: AuditLogger
+): Array<SavedObject<RawAlert>> {
+  return rules.map((rule) => transformRuleForExport(rule as SavedObject<RawAlert>, auditLogger));
+}
+
+function transformRuleForExport(
+  rule: SavedObject<RawAlert>,
+  auditLogger?: AuditLogger
+): SavedObject<RawAlert> {
+  auditLogger?.log(
+    alertAuditEvent({
+      action: AlertAuditAction.EXPORT,
+      savedObject: { type: 'alert', id: rule.id },
+    })
+  );
```

Comment:
It feels wrong to have the type owners manually log the export from the export hooks, but I guess as it's an hidden type, we don't have much choice @elastic/kibana-security?

Reply:
I did this based off this [this comment](https://github.com/elastic/kibana/issues/82027#issuecomment-738493509) and the responses to the comment. If there's a better way to do this please let me know!

Reply:
We have two options available:

1) Rely on the security plugin's default audit logging, and remove the auditing from export hooks. This means that the audit events will be more generic than the audit events that you normally capture for alerts, but that might be OK.
2) Introduce custom auditing like you have here. As this is currently written, we will end up with duplicate audit records: One from the security plugin, and one from this export hook. We will need to somehow teach the security plugin that it's OK to skip auditing specific events under specific conditions.

Option 1 is less work, but introduces inconsistency in the audit logs. When interacting with Alerts normally, you will see the alert-specific audit events. When exporting or viewing alerts from SOM, you will see the generic saved object audit events.

Option 2 is more work (and more complex), but improves the consistency in the audit logs. Even if we take this approach, we will still have some inconsistency, as the SOM UI will still record the generic audit events.

----

Given that we will have inconsistencies either way, my preference is option 1. I think we can remove all auditing from the export hooks here, and just rely on the generic auditing that already exists.

Reply:
@legrego @pgayvallet Sounds good! Removed auditing here https://github.com/elastic/kibana/pull/98802/commits/df73b5f483f1ce21a0b55c7dc70f81fb50821bc9